src/Configuration/Dotenv.hs

```diff
@@ -103,19 +109,35 @@ applySetting kv@(k, v) = do
 -- environment.
 info :: MonadIO m => (String, String) -> DotEnv m ()
 info (key, value) = do
-  Config { .. } <- ask
-  when configVerbose
-    $ liftReaderT . liftIO
-    $ putStrLn
-    $ "[INFO]: Load env '" ++ key ++ "' with value '" ++ value ++ "'"
+  Config {..} <- ask
+  when configVerbose $
+    liftReaderT . liftIO $
+    putStrLn $ "[INFO]: Load env '" ++ key ++ "' with value '" ++ value ++ "'"
 
 -- | The helper allows to avoid exceptions in the case of missing files and
 -- perform some action instead.
 --
 -- @since 0.3.1.0
-onMissingFile
-  :: MonadCatch m
+onMissingFile ::
+     MonadCatch m
   => m a -- ^ Action to perform that may fail because of missing file
-  -> m a               -- ^ Action to perform if file is indeed missing
+  -> m a -- ^ Action to perform if file is indeed missing
   -> m a
 onMissingFile f h = catchIf isDoesNotExistError f (const h)
+
+-- | The helper throws an exception if the allow duplicate is set to False.
+onForbidDuplicates :: MonadIO m => String -> m ()
+onForbidDuplicates key =
+  throw $
+  userError $
+  "[ERROR]: Env '" ++
+  key ++
+  "' is duplicated in a dotenv file. Please, fix that (or remove --no-dups)."
+
+lookUpDuplicates :: MonadIO m => [String] -> m ()
+lookUpDuplicates [] = return ()
+lookUpDuplicates [_] = return ()
+lookUpDuplicates (x:xs) =
+  if x `elem` xs
```

Comment:
@FranzGB I think this will eventually call `onForbidDuplicates` on any env var when `--no-dups` is given. I see you're checking if the env is in the list of env vars. If so, then call `onForbidDuplicates` and that will throw an error. Then, my guess is that even normal files, it will throw an error. Am I wrong? :thinking:

Reply:
You're right @FranzGB Thanks! I got all confused.